src/snowflake/cli/api/entities/common.py

```diff
@@ -1,22 +1,32 @@
 from enum import Enum
-from typing import Generic, TypeVar
+from typing import Generic, TypeVar, get_args
 
 
 class EntityActions(str, Enum):
-    BUNDLE = "bundle"
+    BUNDLE = "action_bundle"
 
 
-TEntityModel = TypeVar("TEntityModel")
+T = TypeVar("T")
 
 
-class EntityBase(Generic[TEntityModel]):
+class EntityBase(Generic[T]):
     """
     Base class for the fully-featured entity classes.
     """
 
-    def __init__(self, entity_model: TEntityModel):
+    def __init__(self, entity_model: T):
         self._entity_model = entity_model
 
+    @classmethod
+    def get_entity_model_type(cls) -> T:
+        """
+        Returns the generic model type specified in each entity class.
+        For example:
+            ApplicationEntity(EntityBase[ApplicationEntityModel]
```

Comment:
- Malformed, please fix.
- Isn't this method intended to return a class object, not an instance? The signature says it returns an instance, but I'm not sure that's true.
- Not sure I get the example. I expect that when I call this on an ApplicationEntity class, it returns ApplicationEntityModel. Is that right? If so, I don't think the example makes that clear at all.

Reply:
Updated